e2e/util/validation.go

```diff
@@ -76,30 +79,48 @@ func GetRamenNameSpace(k8sClient *kubernetes.Clientset) (string, error) {
 
 // IsOpenShiftCluster checks if the given Kubernetes cluster is an OpenShift cluster.
 // It returns true if the cluster is OpenShift, false otherwise, along with any error encountered.
-func IsOpenShiftCluster(k8sClient *kubernetes.Clientset) (bool, error) {
-	discoveryClient := k8sClient.Discovery()
+func IsOpenShiftCluster(k8sClient client.Client) (bool, error) {
+	configList := &unstructured.Unstructured{}
+	configList.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "config.openshift.io",
+		Version: "v1",
+		Kind:    "ClusterVersion",
+	})
 
-	apiGroups, err := discoveryClient.ServerGroups()
-	if err != nil {
-		return false, err
+	err := k8sClient.List(context.TODO(), configList)
+	if err == nil {
+		// found OpenShift only resource type, it is OpenShift
```

Comment:
Is this the official way to detect openshift cluster?

Reply:
There is no official way to detect openshift cluster AFAIK.